### ISSUE TITLE
install error with pylsp dependencies

### ISSUE THREAD

**Issue by tatwww** [albilu/netbeansPython#70]
hi , When I install netbeans python plugin, pylsp dependencies cannot be installed as shown below. Is there any way to solve it?

Alex 

----------------------------------------------------------------------------------------------------------
Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  Getting requirements to build wheel did not run successfully.
  exit code: 1
  
  [20 lines of output]
  Traceback (most recent call last):
    File "C:\Users\AppData\Roaming\NetBeans\18\.pythonlsp\.venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
      main()
    File "C:\Users\AppData\Roaming\NetBeans\18\.pythonlsp\.venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\AppData\Roaming\NetBeans\18\.pythonlsp\.venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
      return hook(config_settings)
             ^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\AppData\Local\Temp\pip-build-env-3wx5ajsq\overlay\Lib\site-packages\setuptools\build_meta.py", line 355, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=['wheel'])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\AppData\Local\Temp\pip-build-env-3wx5ajsq\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in _get_build_requires
      self.run_setup()
    File "C:\Users\AppData\Local\Temp\pip-build-env-3wx5ajsq\overlay\Lib\site-packages\setuptools\build_meta.py", line 507, in run_setup
      super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
    File "C:\Users\AppData\Local\Temp\pip-build-env-3wx5ajsq\overlay\Lib\site-packages\setuptools\build_meta.py", line 341, in run_setup
      exec(code, locals())
    File "<string>", line 7, in <module>
  **UnicodeDecodeError: 'cp950' codec can't decode byte 0xc5 in position 3079: illegal multibyte sequence**
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

Getting requirements to build wheel did not run successfully.
exit code: 1

See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: C:\Users\AppData\Roaming\NetBeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install --upgrade pip

**Comment by albilu** [comment]
please provide details of your environment

**Comment by tatwww** [comment]
Hello, sorry for the late reply. My computer is Windows 10, uses Traditional Chinese, the python version is Python 3.11.4, and netbeans uses version 17.
I tried to install it on another computer but it didn't work. I wonder if it supports a computer whose system language is Chinese, because the warning shows text decoding such as cp950.
Thanks

**Comment by albilu** [comment]
i am not sure if the issue is comming from this plugin.
can you try this command in a terminal and give the result
```bash
pip install "python-lsp-server[all]"
```

**Comment by tatwww** [comment]
hi albilu

I made the following arrangements for this installation

1.
You need to open the terminal with administrator rights to install pip


2. 
If you install pip install "python-lsp-server[all]" with python 3.11.xx, one of the plugins astmonkey 0.3.6 does not seem to support python 3.11.x, so I gave up using python 3.11 and used 3.8 instead. Because astmonkey finally Supported is version 3.8.6.
Finally pip install "python-lsp-server[all]" succeeded

3.
pip install --upgrade setuptools

4.
Installing the plugin using netbeans failed, the following is the error message
Traceback (most recent call last):
   File "C:\Users\AppData\Local\Programs\Python\Python36-32\lib\runpy.py", line 193, in _run_module_as_main
     "__main__", mod_spec)
   File "C:\Users\AppData\Local\Programs\Python\Python36-32\lib\runpy.py", line 85, in _run_code
     exec(code, run_globals)
   File "C:\Users\AppData\Roaming\NetBeans\17\.pythonlsp\.venv\lib\site-packages\pip\__main__.py", line 22, in <module>
     from pip._internal.cli.main import main as _main
   File "C:\Users\AppData\Roaming\NetBeans\17\.pythonlsp\.venv\lib\site-packages\pip\_internal\cli\main.py", line 10, in <module>
     from pip._internal.cli.autocompletion import autocomplete
   File "C:\Users\AppData\Roaming\NetBeans\17\.pythonlsp\.venv\lib\site-packages\pip\_internal\cli\autocompletion.py", line 10, in <module>
     from pip._internal.cli.main_parser import create_main_parser
   File "C:\Users\AppData\Roaming\NetBeans\17\.pythonlsp\.venv\lib\site-packages\pip\_internal\cli\main_parser.py", line 9, in <module>
     from pip._internal.build_env import get_runnable_pip
   File "C:\Users\AppData\Roaming\NetBeans\17\.pythonlsp\.venv\lib\site-packages\pip\_internal\build_env.py", line 15, in <module>
     from pip._vendor.packaging.requirements import Requirement
   File "C:\Users\AppData\Roaming\NetBeans\17\.pythonlsp\.venv\lib\site-packages\pip\_vendor\packaging\requirements.py", line 10, in <module>
     from pip._vendor.pyparsing import ( # noqa
   File "C:\Users\AppData\Roaming\NetBeans\17\.pythonlsp\.venv\lib\site-packages\pip\_vendor\pyparsing\__init__.py", line 125, in <module>
     __version__ = __version_info__.__version__
AttributeError: 'version_info' object has no attribute '__version__'


thanks

**Comment by albilu** [comment]
you lust use python version between 3.8 and 3.11.3 for the plugin

**Comment by tatwww** [comment]
hi 

I tried installing it on another computer. This time I'm using netbeans 18 + python 3.11. I thought it was caused by the system language (Traditional Chinese). Due to some errors it became gibberish.

![image](https://github.com/albilu/netbeansPython/assets/146701714/6e708f22-1eee-451d-abfa-26af332b388f)

**Comment by albilu** [comment]
> hi
> 
> I tried installing it on another computer. This time I'm using netbeans 18 + python 3.11. it that support for windows version netbeans ??
> 
> ![image](https://user-images.githubusercontent.com/146701714/273953373-6e708f22-1eee-451d-abfa-26af332b388f.png)

Wich version of windows are you using here?

**Comment by tatwww** [comment]
> > hi
> > I tried installing it on another computer. This time I'm using netbeans 18 + python 3.11. it that support for windows version netbeans ??
> > ![image](https://user-images.githubusercontent.com/146701714/273953373-6e708f22-1eee-451d-abfa-26af332b388f.png)
> 
> Wich version of windows are you using here?

My first laptop is : 
Version Windows 10 Home Edition
22H2 version
Installed on ‎10/‎10/‎2021
Operating system version 19045.3448
Serial number MP18T3SB
Experience the Windows Feature Experience Kit 1000.19044.1000.0

And the second laptop is : 
Windows 10 Professional

**Comment by albilu** [comment]
this exception shouldnt prevent the plugin of working. it is informative. do you have everything working.
Your configuration should look like this. Otherwise please provide your complete Netbeans log file you will find under 
C:\Users\xxx\AppData\Roaming\Netbeans\18\var\log\messages

![image](https://github.com/albilu/netbeansPython/assets/35330562/d88a18f6-2afb-4e67-a7f8-18bed98c296b)

**Comment by tatwww** [comment]
when i start netbeans 18  it show below error 
![image](https://github.com/albilu/netbeansPython/assets/146701714/f951376f-175d-4aa2-ab9a-6b8f5b2936ce)
![image](https://github.com/albilu/netbeansPython/assets/146701714/535fcfa0-811e-4255-8f38-12bbd39e85d7)
![image](https://github.com/albilu/netbeansPython/assets/146701714/857e9e7f-f55a-4ecb-9d38-1564b9d4862d)


In fact, I don't know what to do next unless I manually use pip install "python-lsp-server[all]"

**Comment by albilu** [comment]
> Idon’t understand why I need to install it again after using pip install successfully and then opening it using netbeans.

it's because it is not installing in the same place. 
do C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install "python-lsp-server[all]" 
and restart netbeans

**Comment by tatwww** [comment]
> > Idon’t understand why I need to install it again after using pip install successfully and then opening it using netbeans.
> 
> it's because it is not installing in the same place. do C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install "python-lsp-server[all]" and restart netbeans

Today, I reinstalled a computer to factory mode and used US as the system program.
This time the software version is as follows:
python 3.10
netbeans 18
windows 10 professional (US)

Today, I reinstalled a computer to factory mode and used US as the system program.
This time the software version is as follows:
python 3.10
netbeans 18
windows 10 professional (US)

This time, the previous garbled code problem did not occur, but a new problem occurred and I used the method you suggested.
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install "python-lsp-server[all]"
and restart netbeans

Summing up the last experience, it seems that this plug-in cannot support the Traditional Chinese language system.

![image](https://github.com/albilu/netbeansPython/assets/146701714/4ad336eb-4715-4281-bbff-c9446c1c2b0b)

**Comment by albilu** [comment]
> used the method you suggested.
> **C:\Users\Tat\AppData\Roaming\Netbeans\18.pythonlsp.venv\Scripts\python.exe -m pip install "python-lsp-server[all]"**
> and restart netbeans

i dont understand. Does the issue solved after following the steps above?

**Comment by tatwww** [comment]
According to the advice you gave me:
C:\Users\Tat\AppData\Roaming\Netbeans\18.pythonlsp.venv\Scripts\python.exe -m pip install "python-lsp-server[all]"
and restart netbeans

The installation cannot be successful, and the result is shown below. In addition, the detailed installation plug-in record is attached in the previous post error.txt
![image](https://github.com/albilu/netbeansPython/assets/146701714/87bc55fc-ce6b-49af-9ea3-6bce4e90dd6e)

**Comment by albilu** [comment]
if C:\Users\Tat\AppData\Roaming\Netbeans\18.pythonlsp.venv\Scripts\python.exe -m pip install "python-lsp-server[all]" 
failed that's mean your issue is comming from your python environment.

The log you attached mentioned a missing Rust Compiler. you are probably missing some dll. i will recommend to fill a ticket for the the https://github.com/python-lsp/python-lsp-server team to ask for help.

**Comment by tatwww** [comment]
Maybe I didn't express myself clearly. When I run C:\Users\Tat\AppData\Roaming\Netbeans\18.pythonlsp.venv\Scripts\python.exe -m pip install "python-lsp-server[all]" in cmd, yes  Successful, but when I restart netbeans, the error.txt content will appear again. As mentioned before, there may be some problems with Windows 10 support because I use two different computers (new systems). Go  The installation was also unsuccessful.  by the way Thank you for your contribution to netbeans. I also like this plug-in. If possible, can I have further contact with you?  I am very curious about the development of this plug-in

**Comment by albilu** [comment]
I just intalled the plugin on a windows 10 without issue. Like i said this seems very likely to come from your environment.From the error_log.txt the issue comes from a sub dependency failing to install on your computer. try the following one by one without netbeans. it will help narrow to the failing dependency:

```
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pyls-isort
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pylsp-mypy
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pylsp-rope
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install black
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install python-lsp-black
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install isort
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pyls-memestra
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install ptpython
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install ipython
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pytest
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pynguin
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install build
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pdoc
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install poetry
```

**Comment by tatwww** [comment]
When I tried to install, I found that only C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pynguin failed to install. The error output is as follows:

____________________________________________________________________________________________________________________________
Collecting astmonkey>=0.3.6
   Using cached astmonkey-0.3.6.tar.gz (10 kB)
   Preparing metadata (setup.py) ... error
   error: subprocess-exited-with-error

   × python setup.py egg_info did not run successfully.
   │ exit code: 1
   ╰─> [6 lines of output]
       Traceback (most recent call last):
         File "<string>", line 2, in <module>
         File "<pip-setuptools-caller>", line 34, in <module>
         File "C:\Users\Tat\AppData\Local\Temp\pip-install-7logw_q2\astmonkey_4d2e3817d288489fa6f7d1a0026b1e9d\setup.py", line 7, in <module>
           long_description = f.read()
       UnicodeDecodeError: 'cp950' codec can't decode byte 0xc5 in position 3079: illegal multibyte sequence
       [end of output]

   note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.


____________________________________________________________________________________________________________________________


C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pyls-isort
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pylsp-mypy
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pylsp-rope
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install black
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install python-lsp-black
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install isort
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pyls-memestra
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install ptpython
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install ipython
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pytest
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pynguin
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install build
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install pdoc
C:\Users\Tat\AppData\Roaming\Netbeans\18\.pythonlsp\.venv\Scripts\python.exe -m pip install poetry


![image](https://github.com/albilu/netbeansPython/assets/146701714/c1757267-2991-416f-9447-db0e14406cda)

In fact, I don’t have a problem with unit test. Is this possible?

**Comment by tatwww** [comment]
Thanks again for your help, I can now use python with netbeans!
![image](https://github.com/albilu/netbeansPython/assets/146701714/44852576-ace8-4b40-8611-f8fe5b0ac4fc)

**Comment by albilu** [comment]
> Actually, I don't need unit tests, will this work? Is it possible to ignore this installation or install it selectively so that netbeans does not need to be installed again when it is reopened?

Yes it will work. Just make sure Auto Update is not checked and the installation procedure will not happen. I added already a change to ignore pynguin in new installation

**Comment by albilu** [comment]
> Also, I would like to ask if there are any code prompts. I don’t seem to find them when using them. For example, when I enter print and if, there are no relevant prompts. Is this normal?

What do you mean i don't understand?

**Comment by tatwww** [comment]
> > Also, I would like to ask if there are any code prompts. I don’t seem to find them when using them. For example, when I enter print and if, there are no relevant prompts. Is this normal?
> 
> What do you mean i don't understand?

I mean is there any python code hint?
